src/client/MetaRequest.h

```diff
@@ -202,9 +202,15 @@ struct MetaRequest {
   bool auth_is_best() {
     if ((head.op & CEPH_MDS_OP_WRITE) || head.op == CEPH_MDS_OP_OPEN ||
         (head.op == CEPH_MDS_OP_GETATTR && (head.args.getattr.mask & CEPH_STAT_RSTAT)) ||
-	head.op == CEPH_MDS_OP_READDIR || send_to_auth) 
+	head.op == CEPH_MDS_OP_READDIR || head.op == CEPH_MDS_OP_MKNOD ||
```

Comment:
Out of curiousity, why  does READDIR go to the auth MDS? Shouldn't any MDS work as well for that?

Reply:
From the MDS code, only the diri's auth MDS could add new dir fragment. Or it will forward the request anyway. The readdir may will add new frag.